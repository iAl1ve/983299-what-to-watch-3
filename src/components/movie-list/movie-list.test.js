import React from "react";
import renderer from "react-test-renderer";
import MovieList from "./movie-list";

const mock = {
  filmsList: [
    {
      title: `Some Title`,
      genre: `Comedy`,
      releaseYear: 2015,
      imgSrc: `Some Path`,
      bgSrc: `iSome Path`,
      posterSrc: `Some Path`,
      ratingScore: 8.7,
      ratingCount: 230,
      description: [
        `Some description`,
      ],
      director: `Some cool directot`,
      starring: [`Actor1`, `Actor2`],
      id: 2,
      videoSrc: `Some Path`,
    },
    {
      title: `Some Title`,
      genre: `Comedy`,
      releaseYear: 2015,
      imgSrc: `Some Path`,
      bgSrc: `iSome Path`,
      posterSrc: `Some Path`,
      ratingScore: 8.7,
      ratingCount: 230,
      description: [
        `Some description`,
      ],
      director: `Some cool directot`,
      starring: [`Actor1`, `Actor2`],
      id: 4,
      videoSrc: `Some Path`,
    }
  ]
};

it(`<MovieList /> should render correctly`, () => {
  const tree = renderer
    .create(
        <MovieList filmsList={mock.filmsList} onMovieCardClick={() => {}} />,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
