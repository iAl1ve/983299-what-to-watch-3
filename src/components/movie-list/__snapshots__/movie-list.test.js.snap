// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="name1"
        height="175"
        src="img/name1.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <a
        className="small-movie-card__link"
        href="/dev-component"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        name1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="name2"
        height="175"
        src="img/name2.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <a
        className="small-movie-card__link"
        href="/dev-component"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        name2
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="name3"
        height="175"
        src="img/name3.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <a
        className="small-movie-card__link"
        href="/dev-component"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        name3
      </a>
    </h3>
  </article>
</div>
`;
